### DISCUSSION TITLE
Please add better typescript.

### DISCUSSION THREAD

**Discussion by KC3NZO** [inertiajs/inertia#984] [Help (React)]
inertia has big problem and that is typescript. for example when I want to use share data in **HandlerInertiaRequest** in laravel  I want to add some extra data and I can't declare typescript interfaces or types.

And big problem is **PageProps** interface with unknown for each key and value, I tried declare interface or but PageProps is not declaring.

Another problem is **useForm** hook in inertia react. some methods and properties like progress, cancel are not exists at all.

Sorry for my bad English I hope some one check this.

**Comment by fmaximus** [comment]
## About Shared properties:

You can declare your shared properties in a .d.ts

Example:
```typescript
interface SharedProps extends PageProps {
    auth: {
        user: {
            id: number,
            name: string,
            email: string
        }
    }
}

export type PageWithSharedProps = InertiaPage<SharedProps>
```

and use it:
```typescript
    const { auth } = usePage<PageWithSharedProps>().props
```

## About the UseForm methods:
I could find progress, but cancel indeed appears to be missing.

**Reply by maungyehtunzaw** [reply]
InertiaPage not define

**Reply by MitichPavel** [reply]
> InertiaPage not define

Hi, you can import it from inertia core like below. Install it before and import in the place where you need.
For install for example use code below:
```
npm install @inertiajs/core
```
and next import
```
import { Page as InertiaPage, PageProps } from '@inertiajs/core';
```
